fuse_core/include/fuse_core/throttled_callback.h

```diff
@@ -145,32 +146,69 @@ class ThrottledCallback
   /**
    * @brief Callback that throttles the calls to the keep callback provided. When dropped because
    * of throttling, the drop callback is called instead.
+   * @details The decision to throttle is made using ros::Time::now when use_wall_time_ is false and ros::WallTime::now
+   * when it is true. The first message after the most recent whole-number multiple of the throttle_period_ is published
+   * and all others are dropped.
    *
    * @param[in] args The input arguments
    */
   template <class... Args>
   void callback(Args&&... args)
+  {
+    const ros::Time now = use_wall_time_ ? ros::Time(ros::WallTime::now().toSec()) : ros::Time::now();
+    callbackImpl(now, std::forward<Args>(args)...);
+  }
+
+  /**
+   * @brief Operator() that simply calls the callback() method forwarding the input arguments
+   *
+   * @param[in] args The input arguments
+   */
+  template <class... Args>
+  void operator()(Args&&... args)
+  {
+    callback(std::forward<Args>(args)...);
+  }
+
+protected:
+  /**
+   * @brief The implementation of the throttled callback
+   *
+   * @param[in] time The time used for throttling
+   * @param[in] args The input arguments
+   */
+  template <class... Args>
+  void callbackImpl(const ros::Time time, Args&&... args)
```

Comment:
Any reason we don't take `ros::Time` by reference?
```suggestion
  void callbackImpl(const ros::Time& time, Args&&... args)
```

Reply:
This is a micro (more like nano!) optimization since sizeof(time) == 8 so it's probably infinitesimally faster to pass by value. Can change to reference since I see that's the convention.